hack/build.sh

```diff
@@ -6,58 +6,27 @@ set -o nounset
 
 script_dir=$(dirname "$(readlink -f "$0")")
 
-registry="${registry:-quay.io/confidential-containers/cloud-api-adaptor-${CLOUD_PROVIDER}}"
-
-supported_arches=(
-	"linux/amd64"
-)
-
-function setup_env_for_arch() {
-	case "$1" in
-		"linux/amd64")
-			kernel_arch="x86_64"
-			;;
-		(*) echo "$1 is not supported" && exit 1
-	esac
-}
+registry="${registry:-quay.io/confidential-containers}"
+name="cloud-api-adaptor-${CLOUD_PROVIDER}"
+tag=$(date +%Y%m%d%H%M%s)
+
+supported_arches="linux/amd64,linux/s390x"
 
 function build_caa_payload() {
 	pushd "${script_dir}/.."
 
-	tag=$(date +%Y%m%d%H%M%s)
-
-	for arch in ${supported_arches[@]}; do
-		setup_env_for_arch "${arch}"
-
-		echo "Building cloud-api-adaptor ${CLOUD_PROVIDER} image for ${arch}"
-		docker buildx build \
-			--build-arg ARCH="${kernel_arch}" \
-			--build-arg CLOUD_PROVIDER="${CLOUD_PROVIDER}" \
-			-f Dockerfile \
-			-t "${registry}:${kernel_arch}-${tag}" \
-			--platform="${arch}" \
-			--load \
-			.
-		docker push "${registry}:${kernel_arch}-${tag}"
-	done
-
-	docker manifest create \
-		${registry}:${tag} \
-		--amend ${registry}:x86_64-${tag}
-
-	docker manifest create \
-		${registry}:latest \
-		--amend ${registry}:x86_64-${tag}
-
-	docker manifest push ${registry}:${tag}
-	docker manifest push --purge ${registry}:latest
+	docker buildx build --platform "${supported_arches}" \
+		--build-arg CLOUD_PROVIDER="${CLOUD_PROVIDER}" \
+		-f Dockerfile \
+		-t "${registry}/${name}:${tag}" \
```

Comment:
How to distinguish the arch specific images ?

Reply:
buildx automatically creates a manifest list with the archs labeled e.g.

```json
{
  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
  "schemaVersion": 2,
  "manifests": [
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:27950a58e2c2b62b569a6de1e74d5ad5b664f864df50934be873d3a9502db256",
      "size": 1091,
      "platform": {
        "architecture": "amd64",
        "os": "linux"
      }
    },
    {
      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
      "digest": "sha256:57563ee8e32ed922819e7ead09e684d14100683c9896a7964dcbe235be73533d",
      "size": 1091,
      "platform": {
        "architecture": "s390x",
        "os": "linux"
      }
    }
  ]
}
```

Reply:
nice..